simfold/src/common/common.cpp

```diff
@@ -649,87 +649,130 @@ PARAMTYPE IL_penalty_by_size_2D (int size1, int size2)
 }
 */
 
+static size_penalties_class *size_penalties;
 
+void create_size_penalties(int nb_nucleotides) {
+    size_penalties = new size_penalties_class(nb_nucleotides);
```

Comment:
generally, new should be avoided in C++11. Instead, use unique_ptr and make_unique. (Here, it does not matter a lot, admittedly)